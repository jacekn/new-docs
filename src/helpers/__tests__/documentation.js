import { buildDocsContents, buildAttributesList } from "../documentation";
import { input, output, attributes } from "./fixtures/documentation";

describe("buildDocsContents", () => {
  it("matches snapshot", () => {
    expect(buildDocsContents(input.basic, "docs")).toStrictEqual(output.basic);
  });
});

describe("buildAttributesList", () => {
  it("Handles empty type fields", () => {
    expect(buildAttributesList(attributes.deeplyNested)).toMatchSnapshot();
  });
  it("Throws with missing whitespace", () => {
    expect(() =>
      buildAttributesList(attributes.throwOnMissingWhitespace),
    ).toThrow("No description found");
  });
  it("parses _links correctly", () => {
    expect(buildAttributesList(attributes.weirdString)).toMatchSnapshot();
  });
});
