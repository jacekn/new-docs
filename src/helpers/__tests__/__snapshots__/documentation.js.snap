// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildAttributesList Handles empty type fields 1`] = `
Array [
  Object {
    "childAttributes": null,
    "description": "DESCRIPTION",
    "name": "ATTRIBUTE",
    "type": "DATA TYPE",
  },
  Object {
    "childAttributes": null,
    "description": "a string description",
    "name": "base attr 1",
    "type": "string",
  },
  Object {
    "childAttributes": Array [
      Object {
        "childAttributes": null,
        "description": "The weight required for a valid transaction including the Allow Trust and Bump Sequence operations.",
        "name": "low_threshold",
        "type": "number",
      },
    ],
    "description": "this one has more attributes",
    "name": "base attr 2",
    "type": "object",
  },
  Object {
    "childAttributes": Array [
      Object {
        "childAttributes": null,
        "description": Object {
          "props": Object {
            "children": Array [
              "top level bold ",
              Object {
                "props": Object {
                  "children": "with code",
                  "mdxType": "inlineCode",
                  "originalType": "inlineCode",
                  "parentName": "strong",
                },
              },
              " in the middle",
            ],
            "mdxType": "strong",
            "originalType": "strong",
            "parentName": "li",
          },
        },
        "name": "attr 1",
        "type": "string",
      },
      Object {
        "childAttributes": null,
        "description": "A hash of characters dependent on the signer type.",
        "name": "attr 2",
        "type": "string",
      },
      Object {
        "childAttributes": Array [
          Object {
            "childAttributes": null,
            "description": "A normal Stellar public key.",
            "name": "child attr 1",
            "type": undefined,
          },
          Object {
            "childAttributes": null,
            "description": Array [
              "some nonsense ",
              Object {
                "props": Object {
                  "children": "with boldness",
                  "mdxType": "em",
                  "originalType": "em",
                  "parentName": "li",
                },
              },
            ],
            "name": "child attr 2",
            "type": undefined,
          },
          Object {
            "childAttributes": null,
            "description": Object {
              "props": Object {
                "children": Array [
                  "top level bold ",
                  Object {
                    "props": Object {
                      "children": "with code",
                      "mdxType": "inlineCode",
                      "originalType": "inlineCode",
                      "parentName": "strong",
                    },
                  },
                  " in the middle",
                ],
                "mdxType": "strong",
                "originalType": "strong",
                "parentName": "li",
              },
            },
            "name": "child attr 3",
            "type": undefined,
          },
        ],
        "description": "this has child attributes",
        "name": "attr 3",
        "type": "string",
      },
    ],
    "description": "some description",
    "name": "base attr 3",
    "type": "array of objects",
  },
]
`;

exports[`buildAttributesList parses _links correctly 1`] = `
Array [
  Object {
    "childAttributes": null,
    "description": "DESCRIPTION",
    "name": "ATTRIBUTES",
    "type": Array [
      "DATA TYPE",
    ],
  },
  Object {
    "childAttributes": null,
    "description": "Provides links for navigating to other pages or to parents and children.",
    "name": "_links",
    "type": Array [
      "array",
    ],
  },
  Object {
    "childAttributes": null,
    "description": "Present when querying an endpoint that responds with a collection of records.",
    "name": "_embedded",
    "type": Array [
      "array",
    ],
  },
]
`;
